src/coremods/core_channel/cmd_topic.cpp

```diff
@@ -38,15 +38,15 @@ CmdResult CommandTopic::HandleLocal(const std::vector<std::string>& parameters,
 	Channel* c = ServerInstance->FindChan(parameters[0]);
 	if (!c)
 	{
-		user->WriteNumeric(Numerics::NoSuchNick(parameters[0]));
+		user->WriteNumeric(ERR_NOSUCHCHANNEL, parameters[0], "No such channel");
 		return CMD_FAILURE;
 	}
 
 	if (parameters.size() == 1)
 	{
 		if ((c->IsModeSet(secretmode)) && (!c->HasUser(user)))
 		{
-			user->WriteNumeric(Numerics::NoSuchNick(c->name));
+			user->WriteNumeric(ERR_NOTONCHANNEL, c->name, "You're not on that channel");
```

Comment:
This should be 'no such channel' to avoid revealing the existence of secret channels..